packages/playwright/src/worker/testInfo.ts

```diff
@@ -410,6 +411,32 @@ export class TestInfoImpl implements TestInfo {
     this._timeoutManager.setIgnoreTimeouts();
   }
 
+  /**
+   * Enables a promise API call to be tracked by the test, alerting if unawaited.
+   *
+   * **NOTE:** Returning from an async function wraps the result in a promise, regardless of whether the return value is a promise. This will automatically mark the promise as awaited. Avoid this.
+   */
+  _wrapPromiseAPIResult<T>(promise: Promise<T>): Promise<T> {
```

Comment:
I'm open to suggestions for better places for this to live.

Reply:
I'd probably add a utility class
```js
class DanglingPromiseScope {
  _danglingCalls: Set
  wrapThennable: ...
  hasCalls()
}
```

and put an instance on TestInfo